src/queues.c

```diff
@@ -183,30 +183,29 @@ int queues_notification_close_id(int id, int reason)
 {
         notification *target = NULL;
 
-        for (GList *iter = g_queue_peek_head_link(displayed); iter;
+        for (GList *iter = g_queue_peek_head_link(displayed);
+             iter && !target;
              iter = iter->next) {
                 notification *n = iter->data;
                 if (n->id == id) {
                         g_queue_remove(displayed, n);
-                        queues_history_push(n);
                         target = n;
-                        break;
                 }
         }
 
-        for (GList *iter = g_queue_peek_head_link(waiting); iter;
+        for (GList *iter = g_queue_peek_head_link(waiting);
+             iter && !target;
              iter = iter->next) {
                 notification *n = iter->data;
                 if (n->id == id) {
                         g_queue_remove(waiting, n);
```

Comment:
Might be worth adding an `assert(target == NULL)` to catch a possible bug where a notification exists in both queues here.

Reply:
I see, this is actually an error and we should use `assert` instead of silently skipping the `waiting` queue.

Reply:
But actually, to make this check valuable, we should not break after assigning `target = n`. We should further traverse the whole notification queue, because then there are no doubled ids.